flow/project.py

```diff
@@ -2693,32 +3245,102 @@ def _guess_cmd(func, name, **kwargs):
             # Extract pre/post conditions and directives from function:
             params = {
                 'pre': pre_conditions.get(func, None),
-                'post': post_conditions.get(func, None),
-                'directives': getattr(func, '_flow_directives', None)}
+                'post': post_conditions.get(func, None)}
 
             # Construct FlowOperation:
             if getattr(func, '_flow_cmd', False):
-                self._operations[name] = FlowOperation(cmd=func, **params)
+                self._operations[name] = FlowCmdOperation(cmd=func, **params)
             else:
-                self._operations[name] = FlowOperation(
-                    cmd=_guess_cmd(func, name, **params), **params)
+                self._operations[name] = FlowOperation(name=name, **params)
                 self._operation_functions[name] = func
 
+    @classmethod
+    def make_group(cls, name, options=None):
+        """Make a FlowGroup named ``name``. And return a decorator to make groups.
```

Comment:
Right now the biggest reason would be to prevent `FlowGroup` from having to deal with modifying user callable's internals, but if refactoring the registration in `flow` goes in the direction I think it should, it makes much more sense to have this class. If we move away from storing values in user callables and instead store the information internally, I believe we should use separate classes to distinguish between the necessary level of information and logic to define and build up operations, groups, directives, etc. in any order before instantiation and the level required to perform the task of that object (create executable commands, request resources, ...).